### PR TITLE
Add "neverending marketing" cheat

### PR DESCRIPTION
**Pull request by octaroot**
This cheat allows the player to have never-ending marketing campaigns

### CODE REVIEW

**Comment by sjoerddebruin** [comment]
W00T!

**Comment by zsilencer** [comment]
the gNeverendingMarketing should also be saved in the network save state in game_load_network/scenario_save_network also might want to increment the NETWORK_STREAM_VERSION

File: src/management/marketing.h [hunk 1]
```diff
@@ -46,7 +46,9 @@ enum{
 extern const money16 AdvertisingCampaignPricePerWeek[6];
 extern uint8 *gMarketingCampaignDaysLeft;
 extern uint8 *gMarketingCampaignRideIndex;
+extern int gNeverendingMarketing;
```
Comment (IntelOrca) [inline src/management/marketing.h]: This doesn't need to be an int, use a bool instead.

**Comment by octaroot** [comment]
> This doesn't need to be an int, use a bool instead.
Also can you define this in cheats.h/c instead and prefix it with gCheats.

Done and done. I was following the weather/climate lock cheat, I guess that one will need refactoring some time in the future.

File: src/management/marketing.c [hunk 2]
```diff
@@ -43,6 +44,14 @@ static const int AdvertisingCampaignGuestGenerationProbabilities[] = { 400, 300,
 uint8 *gMarketingCampaignDaysLeft = RCT2_ADDRESS(0x01358102, uint8);
 uint8 *gMarketingCampaignRideIndex = RCT2_ADDRESS(0x01358116, uint8);
 
+// cheats
+extern void toggle_neverending_marketing();
```
Comment (IntelOrca) [inline src/management/marketing.c]: extern on functions are unnecessary

File: src/management/marketing.c [hunk 3]
```diff
@@ -43,6 +44,14 @@ static const int AdvertisingCampaignGuestGenerationProbabilities[] = { 400, 300,
...
+
+void toggle_neverending_marketing()
```
Comment (IntelOrca) [inline src/management/marketing.c]: I'd move this to cheats

**Comment by octaroot** [comment]
Done and done. This tells me the codebase is in a really bad state, given all of your comments can be applied to other cheats etc.

**Comment by Gymnasiast** [comment]
A lot of the code base reflects the original assembly, with a big refactor planned after full implementation.
As for the cheats: most of them were implemented by the code monkey typing this very message.

**Comment by octaroot** [comment]
I meant no offense by that, this is the first opensource project that I'm contributing to, so my programmer's ego is hurting bad :D I hope non was taken.

**Comment by IntelOrca** [comment]
@octaroot yes, this is true - I have been rather permissive in the past as I didn't want to discourage contributors. But as we are getting very close to removing our dependency on the original binary, I think its good to start organising our new code a bit more to reduce our workload for the **big refactor**.

**Comment by Gymnasiast** [comment]
@octaroot : Don't worry, my remark was meant in a self-deprecating and humorous way. My older code is mostly cheats, and I basically learnt C from working on OpenRCT2.

**Pull request merged**